Detectors/PHOS/simulation/src/PHOSSimulationLinkDef.h

```diff
@@ -17,6 +17,7 @@
 #pragma link C++ class o2::phos::Detector+;
 #pragma link C++ class o2::phos::GeometryParams+;
 #pragma link C++ class o2::base::DetImpl<o2::phos::Detector>+;
+#pragma link C++ class o2::phos::MCLabel+;
```

Comment:
just adding the label is not enough; you also need to add the specialization `...MCLabelContainer<o2::phos::MCLabel>+` since that is the actual type written to disc.